Implemented worker for integration with slack

VeraPochobut opened the pull request:
...based on .net core HostedService; will be registered as windows service.
Logs are processed in memory.

Dump & log analytics settigns moved from db in appsettigs. 
Because we have one web hook for all channels, and we don’t seem to be editing the message pattern for each channel, they are also moved into the settings.

Slack messages are composed by message similarity.
Slack client left as is.

**depends on** #2

VeraPochobut committed:
implemented worker for integration with slack;

moved SlackClient implementation from old project to new;
moved settings to appsettings from db -> removed ef entities;
added top filter for client request.

denisborovnev commented on src/LogAnalyticsViewer.Worker/LogViewer.cs:
```diff
@@ -0,0 +1,56 @@
+        {
+            while (!stoppingToken.IsCancellationRequested)
+            {                
+                using var scope = _scopeFactory.CreateScope();
```
вроде скобки забыли

denisborovnev commented on src/LogAnalyticsViewer.Model/DTO/Event.cs:
```diff
@@ -1,21 +1,33 @@
+        public override bool Equals(object obj) =>
+            obj switch
+            {
+                Event b => b.Message == Message && b.Source == Source,
```
наверное TimeGenerated тоже надо сравнивать

denisborovnev commented on src/LogAnalyticsViewer.Model/Services/Events/EventService.cs:
```diff
@@ -1,63 +1,71 @@
+            {
+                var queryStr = new QueryBuilder()
+                .AddQuery(query)
+                .AddTimeFilter(timeInMinutes)
```
`TimeFilterType.Ago => TimeInMinutes != null ? $"TimeGenerated <= ago({TimeInMinutes}m)" : string.Empty,`

вроде `>=`

еще 
```
public static string ToAzureDate(this DateTime date)
        {
            return @$"datetime(""{ date.ToString("yyyy-MM-dd")}"")";
        }
```
наверное надо фильтровать до того, что приходит в TimeGenerated ??

denisborovnev commented:
еще наверное в message filter надо escape для `"` дописать

VeraPochobut committed:
changes by review

fixed "ago" filter;
more accurate from/to filter (with time)
named TimeFilter format item

VeraPochobut committed:
updated EventUrlFormat
